conda.bzl

```diff
@@ -101,22 +109,33 @@ def _create_conda_build_file(rctx, executable):
 
 def _load_conda_impl(rctx):
     installer = _download_conda(rctx)
-    executable = _install_conda(rctx, installer)
-    _update_conda(rctx, executable)
-    _create_conda_build_file(rctx, executable)
+    conda_executable = _install_conda(rctx, installer)
+    if rctx.attr.install_mamba:
+        _install_mamba(rctx, conda_executable)
+    # TODO: should we update this? The user specified the conda version already.
+    # _update_conda(rctx, conda_executable)
```

Comment:
> I commented out the call to _update_conda. I found it confusing that we specify the version of conda in the build file, but then ask conda to update itself. This seems to go against the "Bazel way" (if I can call it that) of specifying the exact versions for every input to the build process. Can I remove the _update_conda function?

There are hardcoded `miniconda` installers that install a specific version of `conda`. We need that because we need static URLs and SHA256 checksums to be able to download these files and install `conda`.

We want the user to be able to decide which version of `conda` he wants to use (with some default being probably the same version as the downloaded one so updating is cheap as it does nothing). `conda` can update itself so we can use the previously installed `conda` (in whatever version it is) to update itself to the version the user wants.

So I think that `_update_conda` is necessary. What do you think about that now?

Reply:
Also to be clear: `_update_conda` is not about updating to the newest version, but about updating (or even downgrading) to the version specified as a parameter (so it's static so to say in the context of this rule)

Reply:
That makes sense, thanks for explaining. I've updated the diff accordingly.